src/test/regress/spec/isolation_metadata_sync_vs_all.spec

```diff
@@ -0,0 +1,205 @@
+// the test expects to have zero nodes in pg_dist_node at the beginning
+// add 2 nodes and enable metadata syncing to a single node.
+setup
+{
+	SET citus.shard_replication_factor to 1;
+
+	ALTER SEQUENCE pg_catalog.pg_dist_colocationid_seq RESTART 123000;
+	ALTER SEQUENCE pg_catalog.pg_dist_groupid_seq RESTART 123000;
+	ALTER SEQUENCE pg_catalog.pg_dist_node_nodeid_seq RESTART 123000;
+	ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 123000;
+
+	SELECT 1 FROM master_add_node('localhost', 57637);
+	SELECT 1 FROM master_add_node('localhost', 57638);
+
+	-- Create the necessary test utility function
+	CREATE OR REPLACE FUNCTION activate_node_snapshot()
+		RETURNS text[]
+		LANGUAGE C STRICT
+		AS 'citus';
+	SELECT create_distributed_function('activate_node_snapshot()');
+
+	-- Create distributed tables
+	CREATE TABLE ref_table (test_id integer, y int unique);
+	SELECT create_reference_table('ref_table');
+
+	CREATE TABLE dist_table (x int, y int);
+	SELECT create_distributed_table('dist_table', 'x');
+
+	CREATE TABLE dist_partitioned_table (x int, y int) PARTITION BY RANGE(y);
+	SELECT create_distributed_table('dist_partitioned_table', 'x');
+
+	CREATE TABLE dist_partitioned_table_p1(x int, y int);
+}
+
+// ensure neither node's added for the remaining of the isolation tests
+teardown
+{
+	// drop all distributed tables
+	DROP TABLE IF EXISTS ref_table,
+						 dist_table,
+						 dist_partitioned_table,
+						 dist_partitioned_table_p1,
+						 dist_partitioned_table_p2,
+						 new_dist_table,
+						 new_ref_table;
+
+
+	// drop all distributed objects
+	DROP FUNCTION activate_node_snapshot();
+	DROP FUNCTION IF EXISTS squares(int);
+	DROP TYPE IF EXISTS my_type;
+
+	// remove all nodes before the next permutation
+	SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+step "s1-start-metadata-sync"
+{
+	SELECT start_metadata_sync_to_node('localhost', 57638);
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-commit"
+{
+	COMMIT;
+}
+
+step "s2-alter-table"
+{
+	ALTER TABLE dist_table ADD COLUMN z int;
+}
+
+step "s2-add-fk"
+{
+	ALTER TABLE dist_table ADD CONSTRAINT y_fk FOREIGN KEY (y) REFERENCES ref_table(y);
+}
+
+step "s2-drop-fk"
+{
+	ALTER TABLE dist_table DROP CONSTRAINT y_fk;
+}
+
+step "s2-drop-table"
+{
+	DROP TABLE dist_table;
+}
+
+step "s2-create-dist-table"
+{
+	CREATE TABLE new_dist_table(id int, data int);
+	SELECT create_distributed_table('new_dist_table', 'id');
+}
+
+step "s2-create-ref-table"
+{
+	CREATE TABLE new_ref_table(id int, data int);
+	SELECT create_reference_table('new_ref_table');
+}
+
+step "s2-attach-partition"
+{
+	ALTER TABLE dist_partitioned_table ATTACH PARTITION dist_partitioned_table_p1 FOR VALUES FROM (1) TO (9);
```

Comment:
one final request: Can you also please add `CREATE PARTITION OF` test? That goes from a slightly different code-path, so would be good to have that

Reply:
We already cover this. See `s2-create-partition-of` step.